File: BackEnd/app/repositories/points_history_repository.py
from sqlalchemy.exc import IntegrityError
from app.extensions import db
from app.models.points_history_model import PointsHistory


class PointsHistoryRepository:

    def create_history(self, history, commit=True):
        try:
            db.session.add(history)
            if commit:
                db.session.commit()
            else:
                db.session.flush()
            return history, None
        except IntegrityError:
            db.session.rollback()
            return None, "integrity_error"

    def get_history_by_child_id(self, child_id):
        return (
            PointsHistory.query
            .filter_by(child_id=child_id)
            .order_by(PointsHistory.created_at.desc())
            .all()
        )

File: FrontEnd/lib/features/parent/points_history/repositories/points_history_repository.dart
import '../../../../models/points_history_model.dart';
import '../../../../services/points_history_api_service.dart';

class PointsHistoryRepository {
  final PointsHistoryApiService _apiService;

  PointsHistoryRepository({
    PointsHistoryApiService? apiService,
  }) : _apiService = apiService ?? PointsHistoryApiService();

  Future<List<PointsHistoryModel>> getChildHistory(String childId) {
    return _apiService.getChildHistory(childId);
  }
}